File: backend/app/routers/journey.py
"""API router for multi-city journey generation.

V6 Architecture: LLM-First with Iterative Refinement
- Scout: LLM generates initial journey (cities + highlights + travel)
- Enricher: Ground with Google APIs (directions, geocoding)
- Reviewer: Evaluate if humanly feasible
- Planner: Fix issues based on review
- Loop until quality is acceptable
"""

import json
import logging
from datetime import datetime
from typing import Optional

from fastapi import APIRouter, HTTPException, Request

from app.generators.journey_plan.request import JourneyRequest
from fastapi.responses import StreamingResponse
from pydantic import BaseModel, Field

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/api/journey", tags=["journey"])


# ┌───────────────────────────────────────────────────────────────────────────┐
# │ REQUEST MODELS                                                             │
# └───────────────────────────────────────────────────────────────────────────┘

class JourneyPlanRequest(BaseModel):
    """Request to generate a multi-city journey plan."""
    origin: str = Field(..., description="Starting city (e.g., 'Bangalore, India')")
    region: str = Field(..., description="Region to explore (e.g., 'Vietnam', 'South India')")
    destinations: list[str] = Field(default_factory=list, description="Specific cities to visit (optional)")
    total_days: int = Field(..., ge=3, le=30, description="Total trip duration in days")
    start_date: str = Field(..., description="Trip start date (ISO format: YYYY-MM-DD)")
    interests: list[str] = Field(..., min_length=1, description="Travel interests (e.g., ['culture', 'food', 'nature'])")
    pace: str = Field(default="moderate", description="Trip pace: relaxed, moderate, or packed")
    return_to_origin: bool = Field(default=False, description="Whether to return to starting city")
    must_include: list[str] = Field(default_factory=list, description="Cities that must be included")
    avoid: list[str] = Field(default_factory=list, description="Places or types to avoid")
    
    class Config:
        json_schema_extra = {
            "example": {
                "origin": "Bangalore, India",
                "region": "Vietnam",
                "total_days": 14,
                "start_date": "2026-03-01",
                "interests": ["culture", "food", "history", "nature"],
                "pace": "moderate",
                "return_to_origin": False
            }
        }


class DayPlansRequest(BaseModel):
    """Request to generate detailed day plans for an approved journey."""
    journey: dict = Field(..., description="Approved journey plan from /plan/stream")
    start_date: str = Field(..., description="Trip start date (ISO format)")
    interests: list[str] = Field(..., description="Travel interests for activity selection")
    pace: str = Field(default="moderate", description="Daily activity pace")
    travel_mode: str = Field(default="WALK", description="Preferred travel mode in cities: WALK, DRIVE, TRANSIT")
    
    class Config:
        json_schema_extra = {
            "example": {
                "journey": {
                    "theme": "Cultural Vietnam",
                    "total_days": 14,
                    "cities": [
                        {"name": "Hanoi", "country": "Vietnam", "days": 3},
                        {"name": "Ninh Binh", "country": "Vietnam", "days": 2}
                    ],
                    "travel_legs": [
                        {"from_city": "Hanoi", "to_city": "Ninh Binh", "mode": "bus", "duration_hours": 2}
                    ]
                },
                "start_date": "2026-03-01",
                "interests": ["culture", "food", "history"],
                "pace": "moderate"
            }
        }


# ┌───────────────────────────────────────────────────────────────────────────┐
# │ RESPONSE MODELS                                                            │
# └───────────────────────────────────────────────────────────────────────────┘

class JourneyPlanResponse(BaseModel):
    """Response containing the generated journey plan."""
    journey: dict = Field(..., description="Complete journey plan with cities and travel legs")
    review_score: float = Field(..., description="Quality score from review (0-100)")
    iterations: int = Field(..., description="Number of refinement iterations")
    total_travel_hours: float = Field(..., description="Total travel time between cities")
    total_distance_km: float = Field(..., description="Total distance traveled")
    
    class Config:
        json_schema_extra = {
            "example": {
                "journey": {
                    "theme": "Cultural South India",
                    "route": "Bangalore → Mysore → Ooty → Coorg",
                    "total_days": 7,
                    "cities": [{"name": "Mysore", "days": 2}]
                },
                "review_score": 85.0,
                "iterations": 2,
                "total_travel_hours": 12.5,
                "total_distance_km": 450.0
            }
        }


# ┌───────────────────────────────────────────────────────────────────────────┐
# │ JOURNEY PLANNING ENDPOINTS                                                 │
# └───────────────────────────────────────────────────────────────────────────┘

@router.post("/plan", response_model=JourneyPlanResponse)
async def plan_journey(request: JourneyPlanRequest):
    """
    Generate a multi-city journey plan (non-streaming).
    
    For most use cases, prefer `/plan/stream` for real-time progress updates.
    """
    from app.generators.journey_plan import V6Orchestrator
    
    logger.info(f"Journey plan request: {request.origin} → {request.region}")
    
    try:
        # Convert to V6 request format
        orchestrator = V6Orchestrator()
        
        # Build request dict compatible with V6Orchestrator
        v6_request = _build_v6_request(request)
        enriched = await orchestrator.plan_journey(v6_request)
        
        return JourneyPlanResponse(
            journey=_format_journey_response(enriched),
            review_score=0,
            iterations=1,
            total_travel_hours=enriched.total_travel_hours,
            total_distance_km=enriched.total_distance_km,
        )
    except ValueError as e:
        logger.error(f"Validation error: {e}")
        raise HTTPException(status_code=400, detail=str(e))
    except Exception as e:
        logger.error(f"Journey planning failed: {e}", exc_info=True)
        raise HTTPException(status_code=500, detail="Journey planning failed")


@router.post("/plan/stream")
async def plan_journey_stream(request: JourneyPlanRequest, req: Request):
    """
    Stream journey planning progress (recommended).
    
    Events emitted:
    - `scout`: LLM generating initial journey plan
    - `enrich`: Grounding with Google APIs (directions, geocoding)
    - `review`: Evaluating plan feasibility
    - `planner`: Fixing issues based on review
    - `complete`: Final journey plan ready
    - `error`: Processing failed
    
    Each event includes:
    - phase: Current phase name
    - step: Substep (start/progress/complete)
    - message: Human-readable status
    - progress: Overall progress (0-100)
    - iteration: Current iteration number (for refinement loops)
    - data: Phase-specific data (route, cities, etc.)
    """
    from app.generators.journey_plan import V6Orchestrator
    
    logger.info(f"Journey plan stream: {request.origin} → {request.region}")
    
    async def event_generator():
        try:
            orchestrator = V6Orchestrator()
            v6_request = _build_v6_request(request)
            
            async for progress in orchestrator.plan_journey_stream(v6_request):
                event_data = {
                    "phase": progress.phase,
                    "step": progress.step,
                    "message": progress.message,
                    "progress": progress.progress,
                    "iteration": progress.iteration,
                }
                if progress.data:
                    event_data["data"] = progress.data
                
                yield f"event: {progress.phase}\ndata: {json.dumps(event_data)}\n\n"
                
                if progress.phase in ("complete", "error"):
                    break
                    
                if await req.is_disconnected():
                    logger.info("Client disconnected")
                    break
                    
        except Exception as e:
            logger.error(f"Streaming error: {e}", exc_info=True)
            yield f"event: error\ndata: {json.dumps({'error': str(e)})}\n\n"
    
    return StreamingResponse(
        event_generator(),
        media_type="text/event-stream",
        headers={
            "Cache-Control": "no-cache",
            "Connection": "keep-alive",
            "X-Accel-Buffering": "no",
        },
    )


# ┌───────────────────────────────────────────────────────────────────────────┐
# │ DAY PLANS GENERATION ENDPOINTS                                            │
# └───────────────────────────────────────────────────────────────────────────┘

@router.post("/days/stream")
async def generate_day_plans_stream(request: DayPlansRequest, req: Request):
    """
    Generate detailed day-by-day itineraries for each city (streaming).
    
    Takes an approved journey plan and creates detailed itineraries
    for each city using the FastItineraryGenerator.
    
    Events emitted:
    - `city_start`: Beginning generation for a city
    - `city_complete`: Finished city with day plans
    - `city_error`: Error generating city plans (non-fatal)
    - `complete`: All cities done
    - `error`: Fatal error
    
    Each event includes city_name, city_index, total_cities,
    and progress percentages.
    """
    from app.generators.journey_plan import (
        V6DayPlanGenerator,
        JourneyPlan,
        CityStop,
        CityHighlight,
        TravelLeg,
        TransportMode,
    )
    from app.models import Pace, TravelMode as ITravelMode
    
    logger.info(f"Day plans stream for {len(request.journey.get('cities', []))} cities")
    
    async def event_generator():
        try:
            # Reconstruct JourneyPlan from dict
            journey = _parse_journey_dict(request.journey)
            
            # Parse parameters
            start_date = datetime.fromisoformat(request.start_date).date()
            pace = Pace(request.pace.lower())
            travel_mode = ITravelMode(request.travel_mode.upper())
            
            # Generate
            generator = V6DayPlanGenerator()
            
            async for progress in generator.generate_day_plans_stream(
                journey=journey,
                start_date=start_date,
                interests=request.interests,
                pace=pace,
                travel_mode=travel_mode,
            ):
                event_data = {
                    "phase": progress.phase,
                    "city_name": progress.city_name,
                    "city_index": progress.city_index,
                    "total_cities": progress.total_cities,
                    "message": progress.message,
                    "progress": progress.progress,
                    "city_progress": progress.city_progress,
                }
                if progress.data:
                    event_data["data"] = progress.data
                
                yield f"event: {progress.phase}\ndata: {json.dumps(event_data)}\n\n"
                
                if progress.phase in ("complete", "error"):
                    break
                
                if await req.is_disconnected():
                    logger.info("Day plans client disconnected")
                    break
                    
        except Exception as e:
            logger.error(f"Day plans error: {e}", exc_info=True)
            yield f"event: error\ndata: {json.dumps({'error': str(e)})}\n\n"
    
    return StreamingResponse(
        event_generator(),
        media_type="text/event-stream",
        headers={
            "Cache-Control": "no-cache",
            "Connection": "keep-alive",
            "X-Accel-Buffering": "no",
        },
    )


# ┌───────────────────────────────────────────────────────────────────────────┐
# │ HELPER FUNCTIONS                                                           │
# └───────────────────────────────────────────────────────────────────────────┘

def _build_v6_request(request: JourneyPlanRequest) -> JourneyRequest:
    """Convert JourneyPlanRequest to V6Orchestrator format."""
    return JourneyRequest(
        origin=request.origin,
        region=request.region,
        destinations=request.destinations,
        total_days=request.total_days,
        start_date=request.start_date,
        interests=request.interests,
        pace=request.pace,
        return_to_origin=request.return_to_origin,
        must_include=request.must_include,
        avoid=request.avoid,
    )


def _format_journey_response(enriched) -> dict:
    """Format EnrichedPlan to response dict."""
    return {
        "theme": enriched.plan.theme,
        "summary": enriched.plan.summary,
        "route": enriched.plan.route_string,
        "origin": enriched.plan.origin,
        "region": enriched.plan.region,
        "total_days": enriched.plan.total_days,
        "cities": [
            {
                "name": city.name,
                "country": city.country,
                "days": city.days,
                "why_visit": city.why_visit,
                "best_time_to_visit": city.best_time_to_visit,
                "highlights": [
                    {"name": h.name, "description": h.description, "category": h.category}
                    for h in city.highlights
                ],
                "latitude": city.latitude,
                "longitude": city.longitude,
            }
            for city in enriched.plan.cities
        ],
        "travel_legs": [
            {
                "from_city": leg.from_city,
                "to_city": leg.to_city,
                "mode": leg.mode.value,
                "duration_hours": leg.duration_hours,
                "distance_km": leg.distance_km,
                "notes": leg.notes,
                "estimated_cost": leg.estimated_cost,
                "booking_tip": leg.booking_tip,
            }
            for leg in enriched.plan.travel_legs
        ],
    }


def _parse_journey_dict(journey_dict: dict):
    """Parse journey dict back to JourneyPlan object."""
    from app.generators.journey_plan import (
        JourneyPlan,
        CityStop,
        CityHighlight,
        TravelLeg,
        TransportMode,
    )
    
    cities = []
    for c in journey_dict.get("cities", []):
        highlights = []
        for h in c.get("highlights", []):
            if isinstance(h, dict):
                highlights.append(CityHighlight(
                    name=h.get("name", ""),
                    description=h.get("description", ""),
                    category=h.get("category", ""),
                    suggested_duration_hours=h.get("suggested_duration_hours", 2.0),
                ))
            elif isinstance(h, str):
                highlights.append(CityHighlight(name=h, description="", category=""))
        
        cities.append(CityStop(
            name=c.get("name", ""),
            country=c.get("country", ""),
            days=c.get("days", 1),
            highlights=highlights,
            why_visit=c.get("why_visit", ""),
            latitude=c.get("latitude"),
            longitude=c.get("longitude"),
        ))
    
    travel_legs = []
    for leg in journey_dict.get("travel_legs", []):
        mode_str = leg.get("mode", "drive").lower()
        try:
            mode = TransportMode(mode_str)
        except ValueError:
            mode = TransportMode.DRIVE
        
        travel_legs.append(TravelLeg(
            from_city=leg.get("from_city", ""),
            to_city=leg.get("to_city", ""),
            mode=mode,
            duration_hours=leg.get("duration_hours", 0),
            distance_km=leg.get("distance_km"),
            notes=leg.get("notes", ""),
            booking_tip=leg.get("booking_tip"),
        ))
    
    return JourneyPlan(
        theme=journey_dict.get("theme", "Your Journey"),
        summary=journey_dict.get("summary", ""),
        cities=cities,
        travel_legs=travel_legs,
        total_days=journey_dict.get("total_days", sum(c.days for c in cities)),
        origin=journey_dict.get("origin", ""),
        region=journey_dict.get("region", ""),
    )


# Legacy endpoint aliases for backward compatibility
@router.post("/v6/plan", response_model=JourneyPlanResponse, include_in_schema=False)
async def plan_journey_v6_alias(request: JourneyPlanRequest):
    """Alias for /plan (V6 is now the default)."""
    return await plan_journey(request)


@router.post("/v6/plan/stream", include_in_schema=False)
async def plan_journey_v6_stream_alias(request: JourneyPlanRequest, req: Request):
    """Alias for /plan/stream (V6 is now the default)."""
    return await plan_journey_stream(request, req)


@router.post("/v6/days/stream", include_in_schema=False)
async def generate_v6_days_stream_alias(request: DayPlansRequest, req: Request):
    """Alias for /days/stream (V6 is now the default)."""
    return await generate_day_plans_stream(request, req)

File: frontend/src/types/journey.ts
/**
 * TypeScript interfaces for V6 Journey Planning API
 */

import { Pace } from './itinerary';

// ═══════════════════════════════════════════════════════════════
// REQUEST TYPES
// ═══════════════════════════════════════════════════════════════

export type TransportMode = 'flight' | 'train' | 'bus' | 'car' | 'ferry';

/**
 * Request to generate a journey plan.
 * Supports flexible input - destinations, region, or both.
 */
export interface JourneyRequest {
  /** Starting location (e.g., "New York, USA") */
  origin: string;
  
  /** Specific destinations to include (can be empty for LLM suggestions) */
  destinations: string[];
  
  /** Region to explore if destinations not fully specified */
  region?: string;
  
  /** Trip start date (ISO format) */
  start_date: string;
  
  /** Total trip duration (alternative: use end_date) */
  total_days?: number;
  
  /** Trip end date (alternative: use total_days) */
  end_date?: string;
  
  /** User interests for activity selection */
  interests: string[];
  
  /** Daily activity pace */
  pace: Pace;
  
  /** Whether to return to starting city */
  return_to_origin: boolean;
  
  /** Destinations that must be included */
  must_include?: string[];
  
  /** Places or types to avoid */
  avoid?: string[];
  
  /** Preferred modes of transport between cities (optional - AI suggests best for region) */
  transport_preferences?: TransportMode[];
}

// ═══════════════════════════════════════════════════════════════
// V6 JOURNEY TYPES (SCOUT → ENRICH → REVIEW → PLANNER LOOP)
// ═══════════════════════════════════════════════════════════════

/** V6 City Highlight - attraction within a city */
export interface V6CityHighlight {
  name: string;
  description: string;
  category: string;  // food, culture, nature, history, etc.
  suggested_duration_hours?: number;
}

/** V6 Accommodation - hotel for a city stop */
export interface V6Accommodation {
  name: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  place_id?: string;
  rating?: number;
  photo_url?: string;
  price_level?: number;
}

/** V6 City Stop - a destination in the journey */
export interface V6CityStop {
  name: string;
  country: string;
  days: number;
  why_visit: string;
  best_time_to_visit?: string;
  highlights: V6CityHighlight[];
  latitude?: number;
  longitude?: number;
  accommodation?: V6Accommodation;
}

/** V6 Travel Leg - transport between cities */
export interface V6TravelLeg {
  from_city: string;
  to_city: string;
  mode: TransportMode;
  duration_hours: number;
  distance_km?: number;
  notes?: string;
  estimated_cost?: string;
  booking_tip?: string;
  fare?: string;
  num_transfers?: number;
  departure_time?: string;
  arrival_time?: string;
}

/** V6 Journey Plan - complete plan from Scout/Planner */
export interface V6JourneyPlan {
  theme: string;
  summary: string;
  route: string;  // "City1 → City2 → City3"
  origin: string;
  region: string;
  total_days: number;
  cities: V6CityStop[];
  travel_legs: V6TravelLeg[];
  review_score?: number;
  total_travel_hours?: number;
  total_distance_km?: number;
}

/** V6 streaming progress event types */
export type V6EventPhase = 
  | 'scout'
  | 'enrich'
  | 'review'
  | 'planner'
  | 'complete'
  | 'error';

/** V6 streaming progress event */
export interface V6ProgressEvent {
  phase: V6EventPhase;
  step: string;  // 'start', 'complete', 'failed'
  message: string;
  progress: number;  // 0-100
  iteration: number;
  data?: {
    // Scout complete
    route?: string;
    city_names?: string[];  // Scout returns city names as strings
    theme?: string;
    
    // Enrich complete
    total_travel_hours?: number;
    total_distance_km?: number;
    
    // Review complete
    score?: number;
    is_acceptable?: boolean;
    summary?: string;
    critical_issues?: number;
    warnings?: number;
    
    // Planner complete
    revised_route?: string;
    
    // Complete - full journey data
    origin?: string;
    region?: string;
    total_days?: number;
    cities?: V6CityStop[];  // Final plan has full city objects
    travel_legs?: V6TravelLeg[];
    review_score?: number;
    
    // Error
    error?: string;
    
    [key: string]: unknown;
  };
}

// ═══════════════════════════════════════════════════════════════
// V6 DAY PLANS - Detailed itineraries for each city
// ═══════════════════════════════════════════════════════════════

/** Request to generate day plans for a V6 journey */
export interface V6DayPlansRequest {
  journey: V6JourneyPlan;
  start_date: string;  // ISO date string
  interests: string[];
  pace: 'relaxed' | 'moderate' | 'packed';
  travel_mode: 'WALK' | 'DRIVE' | 'TRANSIT';
}

/** Route information between two activities */
export interface V6Route {
  distance_meters: number;
  duration_seconds: number;
  duration_text: string;
  travel_mode: 'WALK' | 'DRIVE' | 'TRANSIT';
  polyline?: string;
}

/** Activity in a day plan */
export interface V6Activity {
  time_start: string;
  time_end: string;
  place: {
    name: string;
    category: string;
    address: string;
    location: { lat: number; lng: number };
    rating?: number;
    website?: string;
    photo_url?: string;
  };
  duration_minutes: number;
  notes?: string;
  route_to_next?: V6Route;
}

/** Single day's itinerary */
export interface V6DayPlan {
  date: string;
  day_number: number;
  theme: string;
  activities: V6Activity[];
  total_walking_minutes?: number;
}

/** Day plans for a single city */
export interface V6CityDayPlans {
  city: string;
  days: number;
  day_plans_count: number;
  error?: string;
}

/** Day plan generation progress event types */
export type V6DayPlanEventPhase = 
  | 'city_start'
  | 'city_complete'
  | 'city_error'
  | 'complete'
  | 'error';

/** Day plan generation progress event */
export interface V6DayPlanProgress {
  phase: V6DayPlanEventPhase;
  city_name: string | null;
  city_index: number;
  total_cities: number;
  city_days: number;  // Number of days for current city
  message: string;
  progress: number;  // 0-100 overall
  city_progress: number;  // 0-100 for current city
  data?: {
    // city_start
    country?: string;
    start_date?: string;
    
    // city_complete
    days_generated?: number;
    day_plans?: Array<{
      date: string;
      day_number: number;
      theme: string;
      activity_count: number;
    }>;
    
    // complete - full results
    journey_theme?: string;
    total_days?: number;
    total_cities?: number;
    city_summaries?: V6CityDayPlans[];
    all_day_plans?: V6DayPlan[];
    
    // error
    error?: string;
    
    [key: string]: unknown;
  };
}

// ═══════════════════════════════════════════════════════════════
// UI CONSTANTS
// ═══════════════════════════════════════════════════════════════

export const TRANSPORT_OPTIONS = [
  { value: 'flight' as TransportMode, label: 'Flights', icon: '✈️', description: 'Travel by air' },
  { value: 'train' as TransportMode, label: 'Trains', icon: '🚄', description: 'Rail travel' },
  { value: 'bus' as TransportMode, label: 'Bus', icon: '🚌', description: 'Coach/bus travel' },
  { value: 'car' as TransportMode, label: 'Drive', icon: '🚗', description: 'Rent a car' },
  { value: 'ferry' as TransportMode, label: 'Ferry', icon: '⛴️', description: 'Boat/ferry travel' },
] as const;

/** Example regions for inspiration (users can type any region) */
export const SUGGESTED_REGIONS = [
  { label: 'Northern Italy', value: 'Northern Italy', example: 'Milan → Venice → Florence' },
  { label: 'Japanese Alps', value: 'Japanese Alps', example: 'Tokyo → Takayama → Kanazawa' },
  { label: 'Swiss Alps', value: 'Swiss Alps', example: 'Zurich → Lucerne → Interlaken' },
  { label: 'Andalusia', value: 'Andalusia, Spain', example: 'Madrid → Seville → Granada' },
  { label: 'Scottish Highlands', value: 'Scottish Highlands', example: 'Edinburgh → Inverness → Isle of Skye' },
  { label: 'Greek Islands', value: 'Greek Islands', example: 'Athens → Santorini → Mykonos' },
  { label: 'Vietnam Coast', value: 'Central Vietnam Coast', example: 'Hanoi → Hue → Hoi An' },
  { label: 'New Zealand South', value: 'South Island New Zealand', example: 'Christchurch → Queenstown → Milford Sound' },
  { label: 'Portugal Coast', value: 'Portuguese Atlantic Coast', example: 'Lisbon → Sintra → Porto' },
  { label: 'Balkans', value: 'Western Balkans', example: 'Dubrovnik → Mostar → Sarajevo' },
] as const;

export const DAYS_OPTIONS = [
  { value: 5, label: '5 days', description: 'Long weekend trip' },
  { value: 7, label: '1 week', description: 'Standard vacation' },
  { value: 10, label: '10 days', description: 'Extended trip' },
  { value: 14, label: '2 weeks', description: 'In-depth exploration' },
  { value: 21, label: '3 weeks', description: 'Epic adventure' },
] as const;

// ═══════════════════════════════════════════════════════════════
// CHAT EDIT TYPES
// ═══════════════════════════════════════════════════════════════

/** Request to edit a journey via chat */
export interface ChatEditRequest {
  /** User's edit request message */
  message: string;
  /** Current journey plan to edit */
  journey: V6JourneyPlan;
  /** Original trip context */
  context?: {
    origin?: string;
    region?: string;
    interests?: string[];
    pace?: string;
  };
}

/** Response from journey chat edit */
export interface ChatEditResponse {
  /** Whether the edit was successfully applied */
  success: boolean;
  /** Assistant's response message */
  message: string;
  /** What the AI understood from the request */
  understood_request: string;
  /** List of changes applied */
  changes_made: string[];
  /** Updated journey plan (if successful) */
  updated_journey: V6JourneyPlan | null;
  /** Error message (if failed) */
  error: string | null;
}

/** Chat message for UI display */
export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: Date;
  changes?: string[];
  isLoading?: boolean;
}

/** Request to edit day plans via chat */
export interface DayPlanChatEditRequest {
  /** User's edit request message */
  message: string;
  /** Current day plans to edit */
  day_plans: V6DayPlan[];
  /** Trip context */
  context?: {
    interests?: string[];
    pace?: string;
  };
}

/** Response from day plan chat edit */
export interface DayPlanChatEditResponse {
  /** Whether the edit was successfully applied */
  success: boolean;
  /** Assistant's response message */
  message: string;
  /** What the AI understood from the request */
  understood_request: string;
  /** List of changes applied */
  changes_made: string[];
  /** Updated day plans (if successful) */
  updated_day_plans: V6DayPlan[] | null;
  /** Error message (if failed) */
  error: string | null;
}
